server/main.py

```diff
@@ -205,3 +205,20 @@ def deleteFlashcard(chatroom_id, front_text, back_text):
 @app.get("/getFlashcards/")
 def getAllFlashcards(chatroom_id):
     return {"result": flashcard.getFlashcards(cur, chatroom_id)}
+
+@app.get("/generateInvite")
+def generateInviteForStudent(target_user_id, chatroom_id):
```

Comment:
make the argument be (sessionid, target_user_email, chatroom_id). then get email using checksessionid. then get the current user student id and uni id by profiles.get_student_uni_id. make sure the call to generate_invite's argument matches.

nit: change function name to generateInvite